vitals/batteries/utils/BatteryHealthCheckUtility.java

```diff
@@ -0,0 +1,36 @@
+package vitals.batteries.utils;
+
+import vitals.batteries.Battery;
+import vitals.batteries.types.LiionBattery;
+
+public class BatteryHealthCheckUtility {
+
+	public static boolean isTemperatureOk(Battery battery) {
+		if(battery.temperature <= battery.minTemperature || battery.temperature >= battery.maxTemperature) {
```

Comment:
similar code check is present in soc too. there is scope of reduction in code duplication.